OSPSuite.Presentation.Importer/Core/IDataSourceFile.cs

```diff
@@ -0,0 +1,40 @@
+﻿using OSPSuite.Infrastructure.Import.Services;
+using System.Collections.Generic;
+
+namespace OSPSuite.Presentation.Importer.Core
+{
+   /// <summary>
+   /// Single file containing the data, e.g. excel file or csv file
+   /// </summary>
+   public interface IDataSourceFile
+   {
+      Dictionary<string, IDataSheet> DataSheets { get; set; }
+	   string Path { get; set; }
+   }
+
+   public abstract class DataSourceFile : IDataSourceFile
+   {
+      protected readonly IImportLogger _logger; //not sure this is the correct logger implementetion
+
+      protected DataSourceFile(IImportLogger logger)
+      {
+         _logger = logger;
+      }
+      
+      public Dictionary<string, IDataSheet> DataSheets 
+      { 
+         get; 
+         set; 
+      }
+
+
+      private string _path;
+      public string Path 
+      {
+         get  => _path; 
+         set { _path = value; DataSheets = LoadFromFile(value); }
```

Comment:
Unexpected side effects of setting the path of a data source. I would call it explicitely

Reply:
I think the logic should remain together. Otherwise undesired behavior could show. Imagine you load a file and then change the path (but do not execute LoadFromFile explicitetly). Then you save everything for reloading next time. Next time you open it you will be opening a completely new file, maybe not even compatible with the configuration you have saved. It will not be able to reload the file definitely. Why would we want to change the path without reloading the file?